### PR TITLE
Remove need for from_url / from_config. Remove now unneeded event_source

### PR DESCRIPTION
**Pull request by maxnoe**
As documented in the data model docs, `__new__` is allowed to return a subclass.

Here I use this to finally enable using `EventSource` like a normal class / every other component.

Users do not need to call `from_url` anymore and Components / Tools do not have to call the special method `from_config` anymore. Everything just works with the `EventSource` constructor.

Thus, I finally removed the now unneeded wrapper function `event_source`. 

As a sidenote: the docstring of `EventSource` was horribly out of date and full of wrong information, please read the current version carefully.

### CODE REVIEW

File: docs/ctapipe_api/reco/ImPACT.rst [hunk 1]
```diff
@@ -98,7 +98,7 @@ examples directory soon.
 
    def reconstruct_ImPACT():
 
-        source = hessio_event_source(filename)
+        source = hessio_EventSource(filename)
```
Comment (watsonjj) [inline docs/ctapipe_api/reco/ImPACT.rst:101]: Probably needs correcting

File: docs/development/code-guidelines.rst [hunk 2]
```diff
@@ -302,7 +302,7 @@ help when writing algorithms:
 
 	 # initialize any algorithms
 
-	 source = calibrated_event_source(filename)
+	 source = calibrated_EventSource(filename)
```
Comment (watsonjj) [inline docs/development/code-guidelines.rst:305]: Probably needs correcting

**Review by watsonjj — CHANGES_REQUESTED** [review]
Very nice change. Theres just two parts that seem to have been unintentionally renamed, and might need manual correction.

File: docs/ctapipe_api/reco/ImPACT.rst (hunk 1 above)
Comment (maxnoe) [reply docs/ctapipe_api/reco/ImPACT.rst:101]: That needs removal, it is outdated docs.

File: docs/development/code-guidelines.rst (hunk 2 above)
Comment (maxnoe) [reply docs/development/code-guidelines.rst:305]: Also outdated docs

File: docs/ctapipe_api/reco/ImPACT.rst (hunk 1 above)
Comment (maxnoe) [reply docs/ctapipe_api/reco/ImPACT.rst:101]: Wait. This looks like an accidentally commited file. Shouldn't the API be autogenerated and not be included in git?

Comment (maxnoe) [reply docs/ctapipe_api/reco/ImPACT.rst:101]: Actually it does not really matter. Event If I replaced `hessio_event_source` with `EventSource`, it wouldn't fix the docs example. And I don't see myself in the position to fix the impact docs in this PR.

**Review by watsonjj — DISMISSED** [review]

File: ctapipe/io/tests/test_event_source.py [hunk 3]
```diff
@@ -97,14 +96,14 @@ def test_from_config_invalid_type():
...
 
 
 def test_event_source_config():
     dataset1 = get_dataset_path("gamma_test_large.simtel.gz")
     dataset2 = get_dataset_path("gamma_test_large.simtel.gz")
     config = Config({"EventSource": {"input_url": dataset1}})
-    reader = event_source(dataset2, config=config)
+    reader = EventSource(dataset2, config=config)
     assert isinstance(reader, SimTelEventSource)
     assert str(reader.input_url) == dataset2
```
Comment (dneise) [inline ctapipe/io/tests/test_event_source.py:108]: what is this test doing? dataset1 and 2 are identical, so what does the `assert` statement check?
(Independent of this PR)

Comment (dneise) [reply ctapipe/io/tests/test_event_source.py:108]: moved into #1517

**Review by dneise — APPROVED** [review]
I think this PR really makes things cleaner.

File: ctapipe/io/eventsource.py [hunk 4]
```diff
@@ -9,86 +9,60 @@
...
+    <ctapipe.io.simteleventsource.SimTelEventSource at ...>
 
-    An example of how to use `ctapipe.core.tool.Tool` and `event_source()`
-    can be found in ctapipe/tools/display_dl1.py.
-
-    However if you are not inside a Tool, you can still create an instance and
-    supply an input_url via:
-
-    >>> event_source = EventSource( input_url="/path/to/file")
+    An ``EvenSource`` can also be created through the configuration system,
```
Comment (kosack) [inline ctapipe/io/eventsource.py:40]: minor typo: missing "t"

**Review by kosack — DISMISSED** [review]

**Comment by maxnoe** [comment]
Had to resolve merge conflicts.

**Review by watsonjj — DISMISSED** [review]

**Comment by maxnoe** [comment]
Sometimes automatic merging breaks things... sorry, please review again.

**Review by watsonjj — APPROVED** [review]

**Review by kosack — APPROVED** [review]

**Pull request merged**